packages/jest-resolve-dependencies/src/__tests__/dependency_resolver.test.ts

```diff
@@ -7,20 +7,24 @@
  */
 'use strict';
 
-const path = require('path');
-const {normalize} = require('jest-config');
-const {buildSnapshotResolver} = require('jest-snapshot');
-const DependencyResolver = require('../index');
+import path from 'path';
+import {Config} from '@jest/types';
+// @ts-ignore
+import {normalize} from 'jest-config';
```

Comment:
What's it used for? If just to create a config, please use `testUtils` instead

Reply:
Tried `makeProjectConfig` but it gives obscure errors (and still needs `@ts-ignore` because it's still JS as well)

Reply:
Rather that than the dependency on `jest-config`.

And you can convert it to TS if you need to

Reply:
Converted to TS in #7801. Should fix your type errors (and the bug with `makeProjectConfig`)

Reply:
Replaced, but had to mess around with config values quite a bit to get all tests green, and it's spamming tons of `jest-haste-map: Watchman crawl failed. Retrying once with node crawler.`

<details>
  <summary>Diff from previous `normalize` to the current `makeProjectConfig`</summary>

  ```
    - Expected
    + Received

      Object {
        "automock": false,
    -   "bail": 0,
        "browser": false,
    -   "cache": true,
    -   "cacheDirectory": "/tmp/jest_rs",
    -   "changedFilesWithAncestor": false,
    +   "cache": false,
    +   "cacheDirectory": "/tmp/jest-resolve-dependencies-test",
        "clearMocks": false,
    -   "collectCoverage": false,
    -   "collectCoverageFrom": null,
    -   "coverageDirectory": "/home/seckinger/proj/jest/coverage",
    -   "coveragePathIgnorePatterns": Array [
    -     "/node_modules/",
    -   ],
    -   "coverageReporters": Array [
    -     "json",
    -     "text",
    -     "lcov",
    -     "clover",
    -   ],
    -   "coverageThreshold": null,
    -   "cwd": "/home/seckinger/proj/jest",
    -   "dependencyExtractor": null,
    +   "coveragePathIgnorePatterns": Array [],
    +   "cwd": "/test_root_dir/",
    +   "detectLeaks": false,
    +   "detectOpenHandles": false,
    +   "displayName": undefined,
        "errorOnDeprecated": false,
    -   "expand": false,
    +   "extraGlobals": Array [],
        "filter": null,
        "forceCoverageMatch": Array [],
        "globalSetup": null,
        "globalTeardown": null,
        "globals": Object {},
        "haste": Object {
    -     "computeSha1": false,
          "providesModuleNodeModules": Array [],
        },
    -   "json": undefined,
    -   "maxConcurrency": 5,
    -   "maxWorkers": 7,
        "moduleDirectories": Array [
          "node_modules",
        ],
        "moduleFileExtensions": Array [
          "js",
    -     "json",
    -     "jsx",
    -     "ts",
    -     "tsx",
    -     "node",
        ],
    -   "moduleNameMapper": Object {},
    +   "moduleLoader": "/test_module_loader_path",
    +   "moduleNameMapper": Array [],
        "modulePathIgnorePatterns": Array [],
    -   "name": "6c1b8fd286c221465246da3710646548",
    -   "noStackTrace": false,
    -   "nonFlagArgs": undefined,
    -   "notify": false,
    -   "notifyMode": "failure-change",
    -   "preset": null,
    +   "modulePaths": Array [],
    +   "name": "test_name",
        "prettierPath": "prettier",
    -   "projects": null,
        "resetMocks": false,
        "resetModules": false,
        "resolver": null,
        "restoreMocks": false,
    -   "rootDir": "/home/seckinger/proj/jest",
    +   "rootDir": ".",
        "roots": Array [
    -     "/home/seckinger/proj/jest/packages/jest-resolve-dependencies",
    +     "./packages/jest-resolve-dependencies",
        ],
    -   "runTestsByPath": false,
        "runner": "jest-runner",
        "setupFiles": Array [],
        "setupFilesAfterEnv": Array [],
        "skipFilter": false,
    +   "skipNodeResolution": false,
    +   "snapshotResolver": null,
        "snapshotSerializers": Array [],
    -   "testEnvironment": "/home/seckinger/proj/jest/packages/jest-environment-jsdom/build/index.js",
    +   "testEnvironment": "node",
        "testEnvironmentOptions": Object {},
    -   "testFailureExitCode": 1,
        "testLocationInResults": false,
    -   "testMatch": Array [
    -     "**/__tests__/**/*.[jt]s?(x)",
    -     "**/?(*.)+(spec|test).[tj]s?(x)",
    +   "testMatch": Array [],
    +   "testPathIgnorePatterns": Array [],
    +   "testRegex": Array [
    +     "\\.test\\.js$",
        ],
    -   "testPathIgnorePatterns": Array [
    -     "/node_modules/",
    -   ],
    -   "testPathPattern": "",
    -   "testRegex": Array [],
    -   "testResultsProcessor": null,
    -   "testRunner": "/home/seckinger/proj/jest/packages/jest-jasmine2/build/index.js",
    -   "testURL": "http://localhost",
    +   "testRunner": "jest-jasmine2",
    +   "testURL": "",
        "timers": "real",
    -   "transform": Array [
    -     Array [
    -       "^.+\\.[jt]sx?$",
    -       "/home/seckinger/proj/jest/packages/babel-jest/build/index.js",
    -     ],
    -   ],
    -   "transformIgnorePatterns": Array [
    -     "/node_modules/",
    -   ],
    -   "updateSnapshot": "new",
    -   "useStderr": false,
    -   "verbose": null,
    -   "watch": false,
    +   "transform": Array [],
    +   "transformIgnorePatterns": Array [],
    +   "unmockedModulePathPatterns": null,
        "watchPathIgnorePatterns": Array [],
    -   "watchman": true,
      }
  ```
</details>

Reply:
Can probably disable `watchman`?

Reply:
Oh right in `createContext` there's `options.watchman`. Was too focused on the config :sweat_smile: